### ISSUE TITLE
Migrating 7.14.0 to 8.1.1 fails with duplicate aliases on data type prevalues

### ISSUE THREAD

**Issue by jdphenix** [umbraco/Umbraco-CMS#6062]
Database migration fails in Umbraco.Core.Migrations.Upgrade.V_8_0_0.DataTypeMigration on line 79 with a `System.ArgumentException: An item with the same key has already been added`. 

    var config = migrator.GetConfiguration(dataType.NodeId, dataType.EditorAlias, group.ToDictionary(x => x.Alias, x => x));

Clearly the migration is not expecting there to be duplicate aliases on the same data type for prevalues, but my production database has this for one data type. Is this a bad assumption of the part of the migration code or do I have a manual data fix in my future? 
       
--

Offending data in cmsDataTypePreValues

id	datatypeNodeId	value	sortorder	alias
249	2977	NULL	3	disableFolderSelect
253	2977	NULL	3	disableFolderSelect
257	2977	NULL	3	disableFolderSelect
255	2977	NULL	1	multiPicker
251	2977	NULL	1	multiPicker
247	2977	NULL	1	multiPicker
248	2977	0	2	onlyImages
252	2977	0	2	onlyImages
256	2977	0	2	onlyImages
258	2977		4	startNodeId
254	2977		4	startNodeId
250	2977		4	startNodeId

--
Stack trace

System.ArgumentException: An item with the same key has already been added.\r\n

at System.ThrowHelper.ThrowArgumentException(ExceptionResource resource)\r\n

at System.Collections.Generic.Dictionary`2.Insert(TKey key, TValue value, Boolean add)\r\n

at System.Linq.Enumerable.ToDictionary[TSource,TKey,TElement](IEnumerable`1 source, Func`2 keySelector, Func`2 elementSelector, IEqualityComparer`1 comparer)\r\n

at Umbraco.Core.Migrations.Upgrade.V_8_0_0.DataTypeMigration.Migrate() in d:\\a\\1\\s\\src\\Umbraco.Core\\Migrations\\Upgrade\\V_8_0_0\\DataTypeMigration.cs:line 79\r\n

at Umbraco.Core.Migrations.MigrationBase.Umbraco.Core.Migrations.IMigration.Migrate() in d:\\a\\1\\s\\src\\Umbraco.Core\\Migrations\\MigrationBase.cs:line 73\r\n

at Umbraco.Core.Migrations.MigrationPlan.Execute(IScope scope, String fromState, IMigrationBuilder migrationBuilder, ILogger logger) in d:\\a\\1\\s\\src\\Umbraco.Core\\Migrations\\MigrationPlan.cs:line 309\r\n

at Umbraco.Core.Migrations.Upgrade.Upgrader.Execute(IScopeProvider scopeProvider, IMigrationBuilder migrationBuilder, IKeyValueService keyValueService, ILogger logger) in d:\\a\\1\\s\\src\\Umbraco.Core\\Migrations\\Upgrade\\Upgrader.cs:line 67\r\n

at Umbraco.Core.Migrations.Install.DatabaseBuilder.UpgradeSchemaAndData(MigrationPlan plan) in d:\\a\\1\\s\\src\\Umbraco.Core\\Migrations\\Install\\DatabaseBuilder.cs:line 498

**Comment by nul800sebastiaan** [comment]
Do you know why the datatype with Id 2977 would have 3 properties with the alias `disableFolderSelect`. It seems like this could not work, how would we know which one to store the value in.

A lot of things have happened over the years in people's databases and we can't anticipate all of the scenario's. This one in particular seems like it would give you problems already. 

It would be good to figure out which one of the entries for each alias has the correct value stored and to delete the rest of them (make backups!).

**Comment by stevemegson** [comment]
I was curious why this wasn't already causing 7.14 to blow up. It looks like the duplicate aliases make it just give up on reading the prevalues, so in effect the values in the database are ignored and the media picker just uses the default configuration.

Saving the data type seems to successfully remove the problem rows if you don't want to poke around in the database by hand.

**Comment by jdphenix** [comment]
I'm working on a local copy of the production database. I lost the insanity necessary to do this sort of thing directly on prod years ago :)

Other than the referenced property being a legacy media picker (😅🐑), no. Is there something about legacy media pickers that would do this? 

I went ahead and deleted the duplicates. I've still got a boatload of obsoleted content and media pickers I've got to figure out how to resolve now, so I probably won't be coming back to this issue for a bit.
